WIP release notes (#2489)
rocky-linux/documentation: docs/release_notes/9_5.md

Later versions of the following Application Streams are now available:

* **Apache HTTP Server 2.4.62**
* **Node.js 22**

### Compilers and development tools

For details on the changes in this category [see the upstream link here](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/9.5_release_notes/index#new-features-compilers-and-development-tools)

Updated system toolchain components:

* **GCC 11.5**
* **Annobin 12.70**

Updated performance tools and debuggers:

* **GDB 14.2**
* **Valgrind 3.23.0**
* **SystemTap 5.1**
* **elfutils 0.191**
* **libabigail 2.5**

Updated performance monitoring tools:

* **PCP 6.2.2**
* **Grafana 10.2.6**

Updated compiler toolsets:

* **GCC Toolset 14**
* **LLVM Toolset 18.1.8**
* **Rust Toolset 1.79.0**
* **Go Toolset 1.22**

Updated default Java implementation to `java-17-openjdk`:  
After this update the `java`, `java-devel` and `jre` packages will install Java 17 instead of Java 11.

### Web console

The `cockpit-files` package got added for easier file management in the web console.

> This file manager can perform basic tasks as browsing the files and directories, copy/move/delete and download/upload files.

Reviewer: ```suggestion
This file manager can perform basic tasks such as browsing the files and directories, copy/move/delete and download/upload files.
```

### Container management

Podman got a major update to v5.0, including new features like:

* Extended Quadlet support (automatic systemd service generation)
* Manifest management
* Full support for multi-architecture image builds

This also includes several new features of the other container tools like Buildah or Skopeo.

## Known issues

As with any release, there are many known issues. Review the [upstream listing](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/9.5_release_notes/index#known-issues-installer-and-image-creation) for the complete details on these issues.

### Help in Anaconda

Currently the help buttons when clicked in Anaconda only show blank pages.

!!! info "Deprecated Software"

>     Some Application Streams, which are part of the Rocky Linux 9.5 repositories, won't receive any further updates. These currently include gcc-toolset 12 (retirement Nov 2024), .NET 6 (retirement Nov 2024), Ruby 3.1 (retirement March 2025) and Node.js 18 (retirement April 2025). Furthermore OpenJDK 11 will reach it's EoL for Rocky Linux 9 in October 2024. Please make sure to use supported versions (gcc-toolset 13 or 14 packages, .NET 8 or 9 packages, the Ruby 3.3 module, Node.js 20 or 22 modules, OpenJDK 1.8.0, 17 or 21 packages) if further patches are desired.

Reviewer: ```suggestion
    Some Application Streams, which are part of the Rocky Linux 9.5 repositories, won't receive any further updates. These currently include gcc-toolset 12 (retirement Nov 2024), .NET 6 (retirement Nov 2024), Ruby 3.1 (retirement March 2025) and Node.js 18 (retirement April 2025). Furthermore OpenJDK 11 will reach its EoL for Rocky Linux 9 in October 2024. Please make sure to use supported versions (gcc-toolset 13 or 14 packages, .NET 8 or 9 packages, the Ruby 3.3 module, Node.js 20 or 22 modules, OpenJDK 1.8.0, 17 or 21 packages) if further patches are desired.
```

* **Apache HTTP Server 2.4.62**
* **Node.js 22**

### Compilers and development tools

For details on the changes in this category [see the upstream link here](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#%23new-features-compilers-and-development-tools)

Updated system toolchain components:

* **GCC 11.5**
* **Annobin 12.70**

Updated performance tools and debuggers:

* **GDB 14.2**
* **Valgrind 3.23.0**
* **SystemTap 5.1**
* **elfutils 0.191**
* **libabigail 2.5**

Updated performance monitoring tools:

* **PCP 6.2.2**
* **Grafana 10.2.6**

Updated compiler toolsets:

* **GCC Toolset 14**
* **LLVM Toolset 18.1.8**
* **Rust Toolset 1.79.0**
* **Go Toolset 1.22**

## Known issues

Many known issues exist in 9.5 including those related to:

* Installer and image creation
* Security
* Software management
* Shells and command-line tools
* Infrastructure services
* Networking
* Kernel
* File systems and storage
* Dynamic programing languages, web and database servers
* Identity management
* Desktop
* Graphics infrastructure
* Virtualization
* Supportability
* Containers

> 

Reviewer 2: ```suggestion

### Help in Anaconda

Currently the help buttons when clicked in Anaconda only show blank pages.

```

A suggestion for @grayeul 's comment.

Review the [upstream listing](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#__WKANCHOR_5g) for the complete details on these issues.

!!! info "Deprecated Software"

>     Some Application Streams, which are part of the Rocky Linux 9.4 repositories, won't receive any further updates. These currently include Node.js 16 (retirement April 2024) and .NET 7 (retirement May 2024). Furthermore OpenJDK 11 will reach it's EoL for Rocky Linux 9 in October 2024. Please make sure to use supported versions (Node.js 18 or 20 modules, .NET 6 or 8 packages, OpenJDK 1.8.0, 17 or 21 packages) if further patches are desired.

Reviewer 2: ```suggestion
    Some Application Streams, which are part of the Rocky Linux 9.5 repositories, won't receive any further updates. These currently include gcc-toolset 12 (retirement Nov 2024), .NET 6 (retirement Nov 2024), Ruby 3.1 (retirement March 2025) and Node.js 18 (retirement April 2025). Furthermore OpenJDK 11 will reach it's EoL for Rocky Linux 9 in October 2024. Please make sure to use supported versions (gcc-toolset 13 or 14 packages, .NET 8 or 9 packages, the Ruby 3.3 module, Node.js 20 or 22 modules, OpenJDK 1.8.0, 17 or 21 packages) if further patches are desired.
```

## Major changes

For a complete list of major changes, see the [upstream listing here](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#__WKANCHOR_6).

Highlights and new features from this release are detailed next.

### Security

> Listed below are security-related highlights in the latest Rocky Linux 9.5 release. For a complete list of security-related changes, see the [upstream link here](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#%23new-features-security).

Reviewer 3: Move to HTML link: https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/9.5_release_notes/new-features#new-features-security

* **SELinux** The SELinux policy now provides a boolean that allows QEMU Guest Agent to execute confined commands 
* **OpenSSL** TLS toolkit upgrades to 3.2.2. OpenSSL now supports certificate compression and adds in Brainpool curves to the TLS 1.3 protocol.
* **ca-certificates** program now provides trusted CA roots in the OpenSSL directory format format
* **crypto-policies** packages are updated to extend its control to algorithm sections in Java
* **NSS** cryptographic toolkit are rebased to upstream version 3.101

### Dynamic programming languages, web, and database servers

> For a detailed list of the changes in this category see the [upstream link here](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#%23technology-previews-dynamic-programming-languages-web-and-database-servers).

Reviewer 3: Move to HTML: https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/9.5_release_notes/new-features#new-features-dynamic-programming-languages-web-and-database-servers

Later versions of the following Application Streams are now available:

* **Apache HTTP Server 2.4.62**
* **Node.js 22**

### Compilers and development tools

> For details on the changes in this category [see the upstream link here](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#%23new-features-compilers-and-development-tools)

Reviewer 3: HTML: https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/9.5_release_notes/new-features#new-features-compilers-and-development-tools

Reviewer 2: Think you took the wrong line but also added this 👍🏻

Updated system toolchain components:

* **GCC 11.5**
* **Annobin 12.70**

Updated performance tools and debuggers:

* **GDB 14.2**
* **Valgrind 3.23.0**
* **SystemTap 5.1**
* **elfutils 0.191**
* **libabigail 2.5**

Updated performance monitoring tools:

* **PCP 6.2.2**
* **Grafana 10.2.6**

Updated compiler toolsets:

* **GCC Toolset 14**
* **LLVM Toolset 18.1.8**
* **Rust Toolset 1.79.0**
* **Go Toolset 1.22**

> ## Known issues

Reviewer 3: I suggest adding a specific callout to the Installer Known Issues, referencing the blank display help:

* In the installer, the Help pages are not displayed properly. The text is there for copy/paste but is invisible

Reviewer 2: And here too 👍🏻

* Supportability
* Containers

> Review the [upstream listing](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/pdf/9.5_release_notes/Red_Hat_Enterprise_Linux-9-9.5_Release_Notes-en-US.pdf#__WKANCHOR_5g) for the complete details on these issues.

Reviewer 3: The link to RedHat Known Issues is wrong.  It should be WKANCHOR_5i  (not 5g)

Reviewer 2: God you covered 👍🏻
